x/params/types/proposal.go

```diff
@@ -0,0 +1,119 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// RouterKey defines the routing key for a ParameterChangeProposal
+	RouterKey = "params"
+
+	// ProposalTypeChange defines the type for a ParameterChangeProposal
+	ProposalTypeChange = "ParameterChange"
+)
+
+// TODO: Finish in subsequent PR
+// Assert ParameterChangeProposal implements proposal.Content at compile-time
+//var _ proposal.Content = ParameterChangeProposal{}
+
+// ParameterChangeProposal defines a proposal which contains multiple parameter
+// changes.
+type ParameterChangeProposal struct {
```

Comment:
curious about what's the rationale of having the ParameterChangeProposal in `x/params` instead of `x/gov`

Reply:
Each module should have it's own governance proposals and handlers defined. Governance will have `Text` and `SoftwareUpgrade` and the `params` module will have this one. This provides the cleanest abstraction and doesn't entangle dependencies imho.